python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -923,6 +923,33 @@ def do_evaluate(
             if maintain_order:
                 return Column(column).sorted_like(values)
             return Column(column)
+        elif self.name == "setsorted":
+            (column,) = (
+                child.evaluate(df, context=context, mapping=mapping)
+                for child in self.children
+            )
+            (asc,) = self.options
+            order = (
+                plc.types.Order.ASCENDING
+                if asc == "ascending"
+                else plc.types.Order.DESCENDING
+            )
+            null_order = plc.types.NullOrder.BEFORE
+            if column.obj.null_count() > 0 and (n := column.obj.size()) > 1:
+                # PERF: This invokes four stream synchronisations!
```

Comment:
Just spitballing a bit here but what about a gather on the first and last index followed by a device to host copy and then inspecting that data from there? Depending on the scenario, we might be able to do that for a bunch of columns at once.

Reply:
We never have more than one column available here, unless we do some significant reworking, because you would have to notice that the list of expressions you're making in the IR node that produces a dataframe are identical (so can be evaluated in a data-parallel fashion over the different columns they reference).

Reply:
So for the gather approach on a single column then, I suppose we'd have a HtoD copy to construct the gather map, the gather itself, then the DtoH copy on the result, that's three right? Am I missing any?

Reply:
I think so. I'm kind of inclined to just leave this for now, and wait for benchmarking.

Reply:
Yeah, it's not blocking by any means, just figured I'd throw an idea in the ring since it seemed like a pain point.